csharp/src/Microsoft.ML.OnnxRuntime/Training/CheckpointState.shared.cs

```diff
@@ -28,7 +28,7 @@ public CheckpointState(string checkpointPath)
         {
             if (NativeTrainingMethods.TrainingEnabled())
             {
-                var envHandle = OrtEnv.Handle; // just so it is initialized
+                var envHandle = OrtEnv.Instance().Handle; // just so it is initialized
```

Comment:
Can we also add this line to the SessionOptions ctor while we're at it? There's a stale PR to do that which never got checked in, and actions like adding an EP to SessionOptions require the Env to be setup so the default logger is valid.

Reply:
Yes, we can